Assets/Scripts/UI/HUDManager.cs

```diff
@@ -0,0 +1,68 @@
+using FMS.TapperRedone.Inventory;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class HUDManager : MonoBehaviour
+{
+
+    [SerializeField] private Image drinkHUDImage;        // Drag your DrinkHUDImage here in the inspector
+    [SerializeField] private Sprite defaultSprite;       // Sprite to show when no drink is held
+
+    private PlayerInventory playerInventory;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        // Assuming PlayerInventory is attached to the player
+        playerInventory = FindObjectOfType<PlayerInventory>();
+
+       // Ensures activation of HUD
+        if (playerInventory != null)
+        {
+            playerInventory.OnDrinkChanged += UpdateDrinkHUD; 
+        }
+
+        // Initialize HUD
+        UpdateDrinkHUD();
```

Comment:
Put into the null check - then you don't have to check null in the UpdateDrinkHUD function itself.

Reply:
adjusted so that Instead of passing null explicitly, checks if the player has a drink and pass the appropriate Item or null during initialization